internal/controller/power_monitor_internal.go

```diff
@@ -102,6 +104,28 @@ func (r *PowerMonitorInternalReconciler) SetupWithManager(mgr ctrl.Manager) erro
 
 	if Config.Cluster == k8s.OpenShift {
 		c = c.Owns(&secv1.SecurityContextConstraints{}, genChanged)
+		c = c.Owns(&corev1.Secret{}, genChanged)
+		c = c.Watches(&corev1.Secret{},
+			handler.EnqueueRequestsFromMapFunc(r.mapSecretToPowerMonitorRequests),
+			builder.WithPredicates(
+				predicate.GenerationChangedPredicate{},
```

Comment:
should we have `GenerationChanged` ?

Reply:
As in is the preference to switch to ResourceVersionChanged (which should cover all changes)?

Reply:
Wait didn't we want to use Generationchanged so we did not have to use Resourceversionchanged which may result in many reconciliations for being to sensitive?

Reply:
No, lets keep it this way .. I didn't realize secrets (certs) may be updated externally. Could you please add a comment why GenerationChanged is needed despite of having AnnotationChanged ?

Reply:
Added the comment